platforms/software/ivy/src/test/groovy/org/gradle/api/publish/ivy/internal/publication/DefaultIvyPublicationTest.groovy

```diff
@@ -401,6 +401,8 @@ class DefaultIvyPublicationTest extends Specification {
     def createArtifact(File file) {
         return Mock(IvyArtifact) {
```

Comment:
I think this is fine for now, but in the future instead of mocking all these Gradle managed types and implementating their properties with `TestUtil.objectFactory().property(...)` we should just have Gradle instantiate the type for us like

```
def createArtifact(File file) {
    def artifact = TestUtil.objectFactory().newInstance(IvyArtifact)
    artifact.file.set(file)
    return artifact
}
```